### PR TITLE
UI/event

### PR DESCRIPTION
**Pull request by Keith-Go**
## Ticket 
- [UI Event screen](https://trello.com/c/5CSEWCmG/29-ui-event-page)

## Video

https://github.com/luke-nguyen-goldenowl/join_me/assets/152830147/966c7ed1-e5ba-4fb7-80a8-941175518bb1

### CODE REVIEW

File: lib/src/features/event/logic/map_page_state.dart [hunk 1]
```diff
@@ -0,0 +1,49 @@
+// ignore_for_file: public_member_api_docs, sort_constructors_first
+import 'package:flutter/foundation.dart';
+import 'package:latlong2/latlong.dart';
+
+class MapPageState {
+  LatLng? currentLocation;
+  bool isLoadingCurrentLocation;
+  List<LatLng> events;
```
Comment (luke-nguyen-goldenowl) [inline lib/src/features/event/logic/map_page_state.dart]: It must be the list of event model, lat and lng are just the params in event model

File: lib/src/features/event/logic/map_page_bloc.dart [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      }
+    }
+  }
+
+  void selectEvent(int event) {
+    emit(state.copyWith(currentEvent: event));
+  }
+
+  void getEvent() {
+    emit(state.copyWith(events: [
```
Comment (luke-nguyen-goldenowl) [inline lib/src/features/event/logic/map_page_bloc.dart]: change events to list event model and mock data

File: lib/src/features/event/logic/type_event.dart [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  TypeEvent type;
+  IconData icon;
+  CategoryEvent({
+    required this.type,
+    required this.icon,
+  });
+}
+
+List<CategoryEvent> categoryEvents = [
+  CategoryEvent(type: TypeEvent.Sports, icon: Icons.sports_basketball),
```
Comment (luke-nguyen-goldenowl) [inline lib/src/features/event/logic/type_event.dart]: create method in enum to get icon

File: lib/src/features/event/view/event_view.dart [hunk 4]
```diff
@@ -1,26 +1,110 @@
...
+class EventHomePage extends StatelessWidget {
+  const EventHomePage({
+    super.key,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocBuilder<EventViewBloc, EventViewState>(
+      buildWhen: (previousState, currentState) {
+        return previousState != currentState;
```
Comment (luke-nguyen-goldenowl) [inline lib/src/features/event/view/event_view.dart]: Change buildWhen for all code in the app, must use the param in state to check, not the entire state

**Comment by Keith-Go** [comment]
resolved

File: lib/src/features/event/logic/type_event.dart [hunk 5]
```diff
@@ -3,18 +3,29 @@ import 'package:flutter/material.dart';
 
 enum TypeEvent { Games, Sports, Music, Movie }
 
+extension TypeEventExtension on TypeEvent {
```
Comment (luke-nguyen-goldenowl) [inline lib/src/features/event/logic/type_event.dart]: Just add the method into the enum, no need to create an extension

File: lib/src/features/event/logic/type_event.dart [hunk 6]
```diff
@@ -3,18 +3,29 @@ import 'package:flutter/material.dart';
...
+  IconData get icon {
+    switch (this) {
+      case TypeEvent.Games:
+        return Icons.games_rounded;
+      case TypeEvent.Sports:
+        return Icons.sports_basketball;
+      case TypeEvent.Music:
+        return Icons.music_note_rounded;
+      case TypeEvent.Movie:
```
Comment (luke-nguyen-goldenowl) [inline lib/src/features/event/logic/type_event.dart]: Always check the default case

**Comment by luke-nguyen-goldenowl** [comment]
After fixing, you can merge

**Pull request merged**